comit/src/expiries.rs

```diff
@@ -0,0 +1,591 @@
+            }
+            // Transitioning to any of the cancel actions takes, be definition, zero time.
+            NoFurtherAction | Abort | WaitForRefund | Refund => Duration::zero(),
+        }
+    }
+
+    /// True if Bob has funded the beta HTLC. We return true as soon as the
+    /// fund transaction has been sent since any time after attempting to refund
+    /// is the correct cancellation action.
+    fn has_funded_beta(&self) -> bool {
+        match self {
+            BobState::Created | BobState::AlphaFunded => false,
+            BobState::FundBetaTransactionSent
+            | BobState::BetaFunded
+            | BobState::RedeemBetaTransactionSeen
+            | BobState::RedeemAlphaTransactionSent
+            | BobState::Done => true,
+        }
+    }
+}
+
+// TODO: Add support for lightning.
+#[derive(Clone, Copy, Debug)]
+pub enum Protocol {
+    Herc20Hbit,
+    HbitHerc20,
+}
+
+impl Protocol {
+    fn config(&self) -> Config {
+        match self {
+            Protocol::Herc20Hbit => Config {
+                alpha_required_confirmations: ethereum_required_confirmations(),
+                beta_required_confirmations: bitcoin_required_confirmations(),
+            },
+            Protocol::HbitHerc20 => Config {
+                alpha_required_confirmations: bitcoin_required_confirmations(),
+                beta_required_confirmations: ethereum_required_confirmations(),
+            },
+        }
+    }
+}
+
+/// Configuration values used during transition period calculations.
+#[derive(Clone, Copy, Debug, PartialEq)]
+pub struct Config {
+    alpha_required_confirmations: usize,
+    beta_required_confirmations: usize,
+}
+
+// TODO: Calculate _real_ values instead of just returning 1 hour.
+impl Config {
+    /// The duration of time it takes for Alice to act.
+    pub fn alice_to_act(&self) -> Duration {
+        1.hours()
+    }
+
+    /// The duration of time it takes for Bob to act.
+    pub fn bob_to_act(&self) -> Duration {
+        1.hours()
+    }
+
+    /// The duration of time it takes for the alpha fund transaction to be
+    /// included in the blockchain.
+    pub fn include_alpha_fund_transaction(&self) -> Duration {
+        1.hours()
+    }
+
+    /// The duration of time it takes for the beta fund transaction to be
+    /// included in the blockchain.
+    pub fn include_beta_fund_transaction(&self) -> Duration {
+        1.hours()
+    }
+
+    /// The duration of time it takes for the alpha redeem transaction to be
+    /// included in the blockchain.
+    pub fn include_alpha_redeem_transaction(&self) -> Duration {
+        1.hours()
+    }
+
+    /// The duration of time it takes for the beta redeem transaction to be
+    /// included in the blockchain.
+    pub fn include_beta_redeem_transaction(&self) -> Duration {
+        1.hours()
+    }
+
+    /// The duration of time it takes for a transaction to reach finality on the
+    /// alpha ledger.
+    pub fn finality_alpha(&self) -> Duration {
```

Comment:
This value is actually a number of blocks. Block mining time may vary.

Once a transaction is confirmed, do you still use this value or do you provide a more exact time depending on the number of remaining confirmations?

Reply:
As per the spike this function represents the minimum time we should wait for a transaction on the alpha ledger to reach finality. It is computed from the number of confirmations and the average block time using a Poisson distribution. See[ open draft PR](https://github.com/comit-network/comit-rs/pull/3058) if you are interested in how I am thinking of doing this.